MyModels/BaseModel.py

```diff
@@ -0,0 +1,46 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        self.id_sequence_counter = 0
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+
+        userid = entry['userid'] # Applicable in Contact.py Only
```

Comment:
`BaseModel` cannot have any child-specific logic. It cannot know anything about the existence of a `user_id` or anything. This whole block should be taken out.